### PR TITLE
No proxy fix

### PR DESCRIPTION
**Pull request by BRONSOLO**
ping @chriddyp - accounts for the `no_proxy` env. var. when setting up the proxy configuration

### CODE REVIEW

File: test/tests.py [hunk 1]
```diff
@@ -104,6 +104,43 @@ def test_huge_request_on_latent_server(self):
...
+        # --> proxy_server and proxy_port should not be set
+        os.environ['no_proxy'] = 'some_url, {}, other_url'.format(stream._url)
+        proxy_server, proxy_port = stream._get_proxy_config()
+        self.assertIsNone(proxy_server)
+        self.assertIsNone(proxy_port)
+
+        # unset env variables
+        os.environ.pop('http_proxy')
+        os.environ.pop('no_proxy')
+
```
Comment (chriddyp) [inline test/tests.py]: thx for the tests!

**Comment by chriddyp** [comment]
:dancer:

File: test/tests.py (hunk 1 above)
Comment (BRONSOLO) [inline test/tests.py]: np!

**Comment by BRONSOLO** [comment]
Thanks.

**Pull request merged**